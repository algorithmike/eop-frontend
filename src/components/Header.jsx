import { connect } from 'react-redux';
// import { Button } from '@material-ui/core';
// import { Link } from 'react-router-dom';
// import MenuIcon from '@material-ui/icons/Menu';

import '../styles/Header.scss';
import { unSetMe } from '../store/slices/me';
import Search from './Search';

const Header = () => {
  // TODO: Implement logout, if needed.
  // const logout = () => {
  //   localStorage.removeItem('eop_auth')
  //   props.unSetMe()
  // }
  return (
    <div className="header">
      <div className="header__menu">
        <img className="logo" alt="EOP logo" src="eop_logo.png"/>
      </div>
      <Search />
      {/* TODO: If added functionality for logged in Users, implement here. */}
      {/* <div className="header__logUser">
        {
            props.me.token 
            ? <Button onClick={logout}>Logout</Button>
            : <Link to="/login">
              <Button>Login</Button>
            </Link>
          }
      </div> */}
    </div>
  )
};

const mapStateToProps = (state) => ({
  me: state.me
})

const mapDispatchToProps = (dispatch) => ({
  unSetMe: (me) => (dispatch(unSetMe(me)))
})

export default connect(mapStateToProps, mapDispatchToProps)(Header); 