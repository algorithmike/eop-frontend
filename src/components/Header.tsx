import React from 'react';
import { Button } from '@material-ui/core';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Eop</h1>
    <div>
      <Link to="/about">
        <Button>About</Button>
      </Link>
      <Link to="/">
        <Button>Home</Button>
      </Link>
      <Link to="/Help">
        <Button>Home</Button>
      </Link>
      <Link to="/results">
        <Button>Test Results Page</Button>
      </Link>
      <Link to="/test">
        <Button>Test 404 Page</Button>
      </Link>
    </div>
  </header>
);

export default Header;