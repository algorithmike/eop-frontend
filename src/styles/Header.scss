@import './themeColors.scss';

.header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0.5rem;
    background-color: $primaryThemeColor;
    border: 1pt solid black;
    top: 0;
    position: sticky;
    z-index: 100;

    .header__menu {
        display: flex;
        align-items: center;
        flex: 0.1;
        
        img {
            height: 8rem;
            object-fit: contain;
            margin-left: 1rem;
        }
    }

    .header__logUser {
        display: flex;
        align-items: center;
        Button {
            color: $primaryThemeColor;
            background-color: black;
            
        }
        Button:hover {
            color: $highlightColor;
        }
    }
}