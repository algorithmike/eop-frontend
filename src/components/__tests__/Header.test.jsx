import { Provider } from 'react-redux';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import renderer from 'react-test-renderer';
import DayjsUtils from '@date-io/dayjs';

import instantiateStore from '../../store/store';
import Header from '../Header';

test('should render header correctly', () => {
    const store = instantiateStore();

    const component = renderer.create(
        <MuiPickersUtilsProvider utils={DayjsUtils}>
            <Provider store={store}>
                <Header />
            </Provider>
        </MuiPickersUtilsProvider>
    );
    const tree = component.toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
})