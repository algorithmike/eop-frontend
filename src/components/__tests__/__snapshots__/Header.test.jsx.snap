// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div
  className="header"
>
  <div
    className="header__menu"
  >
    <img
      alt="EOP logo"
      className="logo"
      src="eop_logo.png"
    />
  </div>
  <div
    className="main"
  >
    <div
      className="main__search"
    >
      <input
        className="filterInput"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        value=""
        variant="outlined"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      className="main__timeFilters"
    >
      <div
        className="MuiFormControl-root MuiTextField-root fromPicker"
        format="MMMM D, YYYY"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
        >
          From
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="January 1, 2021"
          />
        </div>
        
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root toPicker"
        format="MMMM D, YYYY"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
        >
          To
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="March 25, 2021"
          />
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
